### PR TITLE
Fixes #1663 - rewrote LoggedIn and LoggedOut components

### LINKED ISSUE #1663: Fix inconsistent styling of Login tab on Next frontend
**What happened**:
With the merge of #1659 , we had a CSS bug where the Login Tab does have the same styling as the other tabs.
image
**What should have happened**:
They should be same size and have same styling.
**How to reproduce it (as precise as possible)**:
Trigger next front end by running `npm run develop:next` and go to `localhost:8000`

### LINKED ISSUE #1682: Fix onClick reaction area in Next.js MobileHeader
**What happened**: After porting MobileHeader to Next.js, buttons in the side-menu react only when clicked precisely on the text. 

**What should have happened**: They should react when clicked a little bit to the side as well - the onClick area should be increased

**How to reproduce it (as precise as possible)**: Start the Next.js version of the Front-end, open the mobile view, open the side-menu and try clicking not on the text on any of the buttons, but a bit to the side.

**Anything else we need to know?**:

**Environment**:
Have been noticed only in the mobile version, but you might want to check the desktop as well. 
- OS:
- Browser:

### PR DESCRIPTION
**Pull request by izhuravlev**
## Issue This PR Addresses
**Fixes #1663** 
**Also Fixes #1682 for LoggedIn and LoggedOut components**

_Mobile view_
image

_Desktop view_
image

## Type of Change

- [X] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [X] **UI**: Change which improves UI

## Description
Started with fixing inconsistent styling of the Login button component, but ended up fixing both Login and Logout components in both Mobile and Desktop. 

Please take a look at the Logout as well if possible. 

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by HyperTHD — DISMISSED** [review]
LGTM

**Review by humphd — DISMISSED** [review]
Let's get this rebased, but looks good to me.

**Comment by izhuravlev** [comment]
I already got 2 approvals from @HyperTHD and @humphd and rebased, so after all checks complete I will merge this with my Administrator powers

**Pull request merged**
